### PR TITLE
Add support for Android Gradle Plugin 3.5

### PR DESCRIPTION
**Pull request by paragchandra**
These changes simplify our Android-specific logic somewhat, in order to support AGP 3.5+. These changes will break backward compatibility with older versions of AGP, so we will need to version this appropriately and indicate as such in the release notes.

### CODE REVIEW

**Comment by alex-dubrouski** [comment]
So we don’t need AndroidBalancedTest anymore? Nice

**Review by alex-dubrouski — APPROVED** [review]

File: hydra-android-gradle-plugin/README.md [hunk 1]
```diff
@@ -5,7 +5,15 @@ The plugin is responsible for creating a hydra-client, fetching the test blackli
 to the hydra-server.
 
 ##### Here are some caveats of this plugin:
-- If you are using a version of Android Gradle Plugin (AGP) _prior_ to 3.4, you *must also* use a 1.x version of Hydra. Likewise, if you are using AGP 3.4+, you *must also* use Hydra 2.0.0+.
+
+- The following table lists which version of the Hydra Android Plug-in is required for each verison of the Android Gradle Plugin (AGP).
```
Comment (dmajumdar2012) [inline hydra-android-gradle-plugin/README.md]: Please fix `version` spelling.

Comment (paragchandra) [reply hydra-android-gradle-plugin/README.md]: Fixed

**Review by jguerra — APPROVED** [review]
Looks good

**Pull request merged**